### PR TITLE
Bubble sort

### PR DESCRIPTION
**Pull request by 1v4n4**
We created two methods to sort arrays. The first one, bubble_sort returns a sorted array of integers. The second one bubble_sort_by uses yield to input a code block and returned a sorted array of strings

### CODE REVIEW

**Comment by mimipeshy** [comment]
## STATUS: CHANGES REQUESTED 🔴 ♻️ 
Hello team @1v4n4 and @TedLivist 👋🏾 👋🏾 , you have done an amazing job on this milestone 😄 

#### Highlights

1. You have followed the correct git-flow for this milestone ✔️ 
2. You have a decent README file ✔️
3. Your linter checks are okay ✔️  

#### To improve

- [ ] In your README file, after a user has cloned your repo, they will need to `cd` into that folder, kindly add it
- [ ] Two tests are failing and I will highlight them for you  in the comments below ⬇️ 

That's all 😄, kindly submit another review when done
If you need any clarification, kindly post your comment here and tag me @mimipeshy and will be glad to help

Regards,
Peris Ndanu

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      if arr[i] > arr[i + 1]
+        arr[i], arr[i + 1] = arr[i + 1], arr[i]
+      else
+        change = false
+      end
+      i += 1
+    end
+  end
+  arr
+end
```
Comment (mimipeshy) [inline bubble_sort.rb]: - [ ] Bubble_sort should also sort an array in increasing order
``` for example, if I pass the following array [8, 7, 6, 5, 4, 3, 2, 1, 9, 10]  it should sort in this manner [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]```

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      if (yield arr[i], arr[i + 1]).positive?
+        arr[i], arr[i + 1] = arr[i + 1], arr[i]
+      else
+        change = false
+      end
+      i += 1
+    end
+  end
+  arr
+end
```
Comment (mimipeshy) [inline bubble_sort.rb]: - [ ] Bubble_sort_by should sort an array by accepting a block
For example, if I pass
```
   (bubble_sort_by(unsorted) { |left, right| left.length - right.length }).to eq %w[h hi hey hiya hello]
```
 I should get    
```
       expected: ["h", "hi", "hey", "hiya", "hello"]
```

File: bubble_sort.rb (hunk 1 above)
Comment (1v4n4) [reply bubble_sort.rb]: Done!

File: bubble_sort.rb (hunk 2 above)
Comment (1v4n4) [reply bubble_sort.rb]: Done!

**Review by mke2111 — APPROVED** [review]
# 😄 🥇  STATUS: APPROVED  🥇 👏🏽 

## Hey Happy Coders, 👋🏽 
Congratulations, 🥂 . You got minimum requirements for approval. 👍🏽 

### **_Well Done_** 💯 

    1. The tests run just fine
    2. The readme is more descriptive

### **_Room for Improvement_** 🤪 

    1. Add instructions in the readme file on how to sort arrays and procedures to follow

### You may now merge to the main branch and remember to submit a project completion form.
### I wish you the very best in your upcoming projects.

**Cheers 🍹 and Happy coding 💻** 
#
### Best regards
Slack 🆘  Roy Mukuye
Twitter 🐦  [Roy Mukuye](https://twitter.com/Roymkenya)
LinkedIn 🔗  [Roy Mukuye](https://www.linkedin.com/in/roy-mukuye/)

**Pull request merged**